# Profitable App Profiles for the App Store and Google Play Markets

The aim for this project is to find profitable mobile app profiles in the Apple App Store and Google Play Store. The goal is to inform mobile App development and enable the developers to make data-driven decisions with respect to the kind of apps to build.

More specifically, the App development team only build apps that are free to download and install and whose main source of revenue consists of in-app ads. As a result, the revenue for these apps is mostly dependent on the number of users that use the app. The goal for this project is to understand what kinds of apps attract more users.


## Opening and Exploring the Data

As of September 2018, there were approximately 2 million iOS apps available on the App Store, and 2.1 million Android apps on Google Play.

Collecting data for over four million apps requires a significant amount of time and money, so we will analyze a sample of data instead to see whether we can find any relevant existing data at no cost. These are two data sets that seem suitable for this purpose:

* [A data set](https://www.kaggle.com/lava18/google-play-store-apps) containing data about approximately ten thousand Android apps from Google Play. You can download the data set directly from [this link](https://www.kaggle.com/lava18/google-play-store-apps).
* [A data set](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps) containing data about approximately seven thousand iOS apps from the App Store. You can download the data set directly from [this link](https://www.kaggle.com/ramamet4/app-store-apple-data-set-10k-apps).

In [1]:
from csv import reader

### The Google Play data set ###
opened_file = open('googleplaystore.csv')
read_file = reader(opened_file)
android = list(read_file)
android_header = android[0]
android = android[1:]

### The App Store data set ###
opened_file = open('AppleStore.csv')
read_file = reader(opened_file)
ios = list(read_file)
ios_header = ios[0]
ios = ios[1:]

The function **explore_data()** will make it easier to explore rows in each dataset in a more readable way. It can also show the number of rows and columns for any data set.

In [2]:
def explore_data(dataset, start, end, rows_and_columns=False):
    dataset_slice = dataset[start:end]    
    for row in dataset_slice:
        print(row)
        print('\n')
        
    if rows_and_columns:
        print('Number of rows:', len(dataset))
        print('Number of columns:', len(dataset[0]))

**Google Play Store Data**

| Column Name | Definition |
| --- | --- |
| App | Application Name |
| Category | Category app belongs to |
| Rating | Overall user rating |
| Reviews | Number of user reviews |
| Size | Size of the app |
| Installs | Number of user downlaods/installs |
| Type | Paid or Free |
| Price | Price of the app |
| Content Rating | Age group |
| Genres | Additional categories belongs to |
| Last Updated | Date whent he app was last updated |
| Current Ver | Current version of the app |
| Android Ver | Min required Android version |

In [3]:
print(android_header)
print('\n')
explore_data(android, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['Coloring book moana', 'ART_AND_DESIGN', '3.9', '967', '14M', '500,000+', 'Free', '0', 'Everyone', 'Art & Design;Pretend Play', 'January 15, 2018', '2.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


Number of rows: 10841
Number of columns: 13


**iOS App Data**

| Column Name | Definition |
| --- | --- |
| id | App ID |
| track_name | App Name |
| size_bytes | Size (in Bytes) |
| currency | Currency Type |
| price | Price amount |
| rating_count_tot | User Rating counts (for all ver) |
| rating_count_ver | User Rating counts (for current ver) |
| user_rating | Avg User Rating vale (for all ver) |
| user_rating_ver | Avg User Rating value (for current ver) |
| ver | Latest version code |
| cont_rating | Content Rating |
| prime_genre | Primary Genre |
| sup_devices.num | Number of supporting devices |
| ipadSc_urls.num | Number of screenshots displayed |
| lang.num | Number of supported languages |
| vpp_lic | Vpp Device Based Licensing Enabled |

In [4]:
print(ios_header)
print('\n')
explore_data(ios, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows: 7197
Number of columns: 17


## Data Cleaning

* Detect inaccurate data, and correct or remove it.
* Detect duplicate data, and remove duplicates.
* Remove non-English apps
* Remove apps that aren't free

**Removing inaccurate data from the Google Play data set**

In the online discussion, a user has pointed out there is an [error in the row with the index 10472](https://www.kaggle.com/lava18/google-play-store-apps/discussion/66015).

In [5]:
print(android_header)
print('\n')
explore_data(android, 10471, 10473)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Xposed Wi-Fi-Pwd', 'PERSONALIZATION', '3.5', '1042', '404k', '100,000+', 'Free', '0', 'Everyone', 'Personalization', 'August 5, 2014', '3.0.0', '4.0.3 and up']


['Life Made WI-Fi Touchscreen Photo Frame', '1.9', '19', '3.0M', '1,000+', 'Free', '0', 'Everyone', '', 'February 11, 2018', '1.0.19', '4.0 and up']




There is a clear error on line index 10472. The rating is '19'. However, the maximum possible rating on the Google Play Store is a 5. As a result, this line will be deleted.

In [6]:
del android[10472]

**Removing duplicate data from the Google Play data set**

In [7]:
duplicate_apps = []
unique_apps = []

for row in android:
    name = row[0]
    if name in unique_apps:
        duplicate_apps.append(name)
    else:
        unique_apps.append(name)

print('Number of duplicate apps: ', len(duplicate_apps))
print('\n')
print('Examples of duplicates: ', duplicate_apps[:10])

Number of duplicate apps:  1181


Examples of duplicates:  ['Quick PDF Scanner + OCR FREE', 'Box', 'Google My Business', 'ZOOM Cloud Meetings', 'join.me - Simple Meetings', 'Box', 'Zenefits', 'Google Ads', 'Google My Business', 'Slack']


Let's see what is going on with the duplicates by looking at the Slack app.

In [8]:
print(android_header)
print('\n')
for row in android:
    name = row[0]
    if name == 'Slack':
        print(row)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51507', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']
['Slack', 'BUSINESS', '4.4', '51510', 'Varies with device', '5,000,000+', 'Free', '0', 'Everyone', 'Business', 'August 2, 2018', 'Varies with device', 'Varies with device']


The only column that varies is 'Reviews'. We will assume the entry with the most reviews is the most recent recording, so we will only keep the row with the highest number of reviews and remove any other entry for a given app.

Start by creating a dictionary with all apps and their max number of reviews.

In [9]:
max_reviews = {}

for row in android:
    name = row[0]
    n_reviews = int(row[3])
    if name in max_reviews and n_reviews > max_reviews[name]:
        max_reviews[name] = n_reviews
    elif name not in max_reviews:
        max_reviews[name] = n_reviews

Then, remove duplicates by only keeping the app line with maximum number of reviews.

In [10]:
android_unique = []
added = []

for row in android:
    name = row[0]
    n_reviews = int(row[3])
    if name not in added and n_reviews == max_reviews[name]:
        android_unique.append(row)
        added.append(name)

In [11]:
print('Expected Length: ', len(android) - len(duplicate_apps))
print('Actual Length: ', len(android_unique))

Expected Length:  9659
Actual Length:  9659


The actual length of only unique Android apps matches the expected length.

In [12]:
print(android_header)
print('\n')
explore_data(android_unique, 0, 5, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


['Pixel Draw - Number Art Coloring Book', 'ART_AND_DESIGN', '4.3', '967', '2.8M', '100,000+', 'Free', '0', 'Everyone', 'Art & Design;Creativity', 'June 20, 2018', '1.1', '4.4 and up']


['Paper flowers instructions', 'ART_AND_DESIGN', '4.4', '167', '5.6M', '50,000+', 'Free', '0', 'Every

**Keeping English ONLY apps from Google Play and App Store data sets**

This analysis will focus on apps directed toward an English-speaking audience. Thus, all non-English app names will be removed from the data set as those names suggest they are not directed toward an English-speaking audience.

In [13]:
def check_English(string):
    for character in string:
        if ord(character) > 127:
            return False
    return True

print(check_English('Instagram'))
print(check_English('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_English('Docs To Go™ Free Office Suite'))
print(check_English('Instachat 😜'))

True
False
False
False


In [14]:
print(ord('™'))
print(ord('😜'))

8482
128540


To accommodate English apps with characters outside of the standard English characters and to minimize the impact of data loss, the check_English function will be modified to allow for three characters that correspond to numbers that fall outside the ASCII range.

In [15]:
def check_English(string):
    non_ascii = 0
    for character in string:
        if ord(character) > 127:
            non_ascii += 1
    if non_ascii > 3:
        return False
    else:
        return True

print(check_English('Instagram'))
print(check_English('爱奇艺PPS -《欢乐颂2》电视剧热播'))
print(check_English('Docs To Go™ Free Office Suite'))
print(check_English('Instachat 😜'))

True
False
True
True


In [16]:
### The Google Play data set ###
android_english = []

for row in android_unique:
    name = row[0]
    if check_English(name):
        android_english.append(row)

print(android_header)
print('\n')
explore_data(android_english, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 9614
Number of columns: 13


In [17]:
### The App Store data set ###
ios_english = []

for row in ios:
    name = row[2]
    if check_English(name):
        ios_english.append(row)

print(ios_header)
print('\n')
explore_data(ios_english, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['1', '281656475', 'PAC-MAN Premium', '100788224', 'USD', '3.99', '21292', '26', '4', '4.5', '6.3.5', '4+', 'Games', '38', '5', '10', '1']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


Number of rows: 6183
Number of columns: 17


**Keeping FREE Apps ONLY from the Google Play and App Store data set**

In [18]:
### The Google Play data set ###
android_free = []

for row in android_english:
    if row[7] == '0':
        android_free.append(row)

print(android_header)
print('\n')
explore_data(android_free, 0, 3, True)

['App', 'Category', 'Rating', 'Reviews', 'Size', 'Installs', 'Type', 'Price', 'Content Rating', 'Genres', 'Last Updated', 'Current Ver', 'Android Ver']


['Photo Editor & Candy Camera & Grid & ScrapBook', 'ART_AND_DESIGN', '4.1', '159', '19M', '10,000+', 'Free', '0', 'Everyone', 'Art & Design', 'January 7, 2018', '1.0.0', '4.0.3 and up']


['U Launcher Lite – FREE Live Cool Themes, Hide Apps', 'ART_AND_DESIGN', '4.7', '87510', '8.7M', '5,000,000+', 'Free', '0', 'Everyone', 'Art & Design', 'August 1, 2018', '1.2.4', '4.0.3 and up']


['Sketch - Draw & Paint', 'ART_AND_DESIGN', '4.5', '215644', '25M', '50,000,000+', 'Free', '0', 'Teen', 'Art & Design', 'June 8, 2018', 'Varies with device', '4.2 and up']


Number of rows: 8864
Number of columns: 13


In [19]:
### The App Store data set ###
ios_free = []

for row in ios_english:
    if row[5] == '0':
        ios_free.append(row)

print(ios_header)
print('\n')
explore_data(ios_free, 0, 3, True)

['', 'id', 'track_name', 'size_bytes', 'currency', 'price', 'rating_count_tot', 'rating_count_ver', 'user_rating', 'user_rating_ver', 'ver', 'cont_rating', 'prime_genre', 'sup_devices.num', 'ipadSc_urls.num', 'lang.num', 'vpp_lic']


['2', '281796108', 'Evernote - stay organized', '158578688', 'USD', '0', '161065', '26', '4', '3.5', '8.2.2', '4+', 'Productivity', '37', '5', '23', '1']


['3', '281940292', 'WeatherBug - Local Weather, Radar, Maps, Alerts', '100524032', 'USD', '0', '188583', '2822', '3.5', '4.5', '5.0.0', '4+', 'Weather', '37', '5', '3', '1']


['4', '282614216', 'eBay: Best App to Buy, Sell, Save! Online Shopping', '128512000', 'USD', '0', '262241', '649', '4', '4.5', '5.10.0', '12+', 'Shopping', '37', '5', '9', '1']


Number of rows: 3222
Number of columns: 17


## Analysis


After cleaning the data, 8864 Android apps and 3222 iOS apps are available for analysis.

The aim is to develop and add an app on both Google Play and the App Store.

To minimize risks and overhead, our validation strategy for an app idea is comprised of three steps:

1. Build a minimal Android version of the app, and add it to Google Play.
2. If the app has a good response from users, develop it further.
3. If the app is profitable after six months, build an iOS version of the app and add it to the App Store.

Therefore, the goal is to find app profiles that are successful on both markets. To start, we'll look at a frequency table for the prime_genre column of the App Store data set, and the Genres and Category columns of the Google Play data set.

### Functions

* The function **freq_table()** generates frequency tables that show percentages.

* The function **display_table()** displays the percentages fo a frequency table in descending order.

In [20]:
def freq_table(dataset, index):
    frequency_table = {}
    total = 0
    for row in dataset:
        total += 1
        if row[index] in frequency_table:
            frequency_table[row[index]] += 1
        else:
            frequency_table[row[index]] = 1
    
    percent_table = {}
    for each in frequency_table:
        percent_table[each] = frequency_table[each] / total * 100
    
    return percent_table

def display_table(dataset, index):
    table = freq_table(dataset, index)
    table_display = []
    for key in table:
        key_val_as_tuple = (table[key], key)
        table_display.append(key_val_as_tuple)

    table_sorted = sorted(table_display, reverse = True)
    for entry in table_sorted:
        print(entry[1], ':', entry[0])

#### Google Play

In [21]:
### Google Play data set - Frequency Table on Category ###
display_table(android_free,1)

FAMILY : 18.907942238267147
GAME : 9.724729241877256
TOOLS : 8.461191335740072
BUSINESS : 4.591606498194946
LIFESTYLE : 3.9034296028880866
PRODUCTIVITY : 3.892148014440433
FINANCE : 3.7003610108303246
MEDICAL : 3.531137184115524
SPORTS : 3.395758122743682
PERSONALIZATION : 3.3167870036101084
COMMUNICATION : 3.2378158844765346
HEALTH_AND_FITNESS : 3.0798736462093865
PHOTOGRAPHY : 2.944494584837545
NEWS_AND_MAGAZINES : 2.7978339350180503
SOCIAL : 2.6624548736462095
TRAVEL_AND_LOCAL : 2.33528880866426
SHOPPING : 2.2450361010830324
BOOKS_AND_REFERENCE : 2.1435018050541514
DATING : 1.861462093862816
VIDEO_PLAYERS : 1.7937725631768955
MAPS_AND_NAVIGATION : 1.3989169675090252
FOOD_AND_DRINK : 1.2409747292418771
EDUCATION : 1.1620036101083033
ENTERTAINMENT : 0.9589350180505415
LIBRARIES_AND_DEMO : 0.9363718411552346
AUTO_AND_VEHICLES : 0.9250902527075812
HOUSE_AND_HOME : 0.8235559566787004
WEATHER : 0.8009927797833934
EVENTS : 0.7107400722021661
PARENTING : 0.6543321299638989
ART_AND_DESIGN : 

In [22]:
### Google Play data set - Frequency Table on Genres ###
display_table(android_free,-4)

Tools : 8.449909747292418
Entertainment : 6.069494584837545
Education : 5.347472924187725
Business : 4.591606498194946
Productivity : 3.892148014440433
Lifestyle : 3.892148014440433
Finance : 3.7003610108303246
Medical : 3.531137184115524
Sports : 3.463447653429603
Personalization : 3.3167870036101084
Communication : 3.2378158844765346
Action : 3.1024368231046933
Health & Fitness : 3.0798736462093865
Photography : 2.944494584837545
News & Magazines : 2.7978339350180503
Social : 2.6624548736462095
Travel & Local : 2.3240072202166067
Shopping : 2.2450361010830324
Books & Reference : 2.1435018050541514
Simulation : 2.0419675090252705
Dating : 1.861462093862816
Arcade : 1.8501805054151623
Video Players & Editors : 1.7712093862815883
Casual : 1.7599277978339352
Maps & Navigation : 1.3989169675090252
Food & Drink : 1.2409747292418771
Puzzle : 1.128158844765343
Racing : 0.9927797833935018
Role Playing : 0.9363718411552346
Libraries & Demo : 0.9363718411552346
Auto & Vehicles : 0.9250902527075

While the difference between the Genres and the Category columns is not clear, it is evident that the Genres column has more categories than Category. Since the goal is to look at the bigger picture, moving forward this analysis will work with Category column.

With regards to the Category column, 'Family' is a little ambiguous. Upon further investigation, it appears the family category is mostly games for kids, which would bring total gaming apps up to 28%.

Overall, most apps seemed to be designed for practical purposes (tools, business, lifestyle, productivity, etc.)

#### App Store

In [23]:
### App Store data set - Frequency Table on Prime Genres ###
display_table(ios_free,-5)

Games : 58.16263190564867
Entertainment : 7.883302296710118
Photo & Video : 4.9658597144630665
Education : 3.662321539416512
Social Networking : 3.2898820608317814
Shopping : 2.60707635009311
Utilities : 2.5139664804469275
Sports : 2.1415270018621975
Music : 2.0484171322160147
Health & Fitness : 2.0173805090006205
Productivity : 1.7380509000620732
Lifestyle : 1.5828677839851024
News : 1.3345747982619491
Travel : 1.2414649286157666
Finance : 1.1173184357541899
Weather : 0.8690254500310366
Food & Drink : 0.8069522036002483
Reference : 0.5586592178770949
Business : 0.5276225946617008
Book : 0.4345127250155183
Navigation : 0.186219739292365
Medical : 0.186219739292365
Catalogs : 0.12414649286157665


Among the free English apps, more than half are games, followed by Entertainment apps and photo and video apps. It seems the App Store for free English apps is dominated by apps that are designed for fun (games, entertainment, photo and video, social networking, sports, music, etc.), while apps with practical purposes (education, shopping, utilities, productivity, lifestyle, etc.) are more rare.

However, note that just because "fun apps" are the most numerous is not an indicator that they also have the greatest number of users.

Up to this point, it seems the **App Store** is dominated by apps designed for fun, while the **Google Play store** shows a more balanced distribution of both practical and fun apps. But most importantly, we need to discover what kind of apps have the most users.

## Most Popular Apps by Genre
To find out what genres are the most popular (have the most users), we can calculate the average number of installs for each app genre. The Google play data set provides this information in the `Install` column. In the App Store data set, we can use the total number of user ratings as a proxy from the `rating_count_tot` column.

#### Google Play Store - Avg number of installs per app genre

In [24]:
display_table(android_free, 5) # Installs column

1,000,000+ : 15.726534296028879
100,000+ : 11.552346570397113
10,000,000+ : 10.548285198555957
10,000+ : 10.198555956678701
1,000+ : 8.393501805054152
100+ : 6.915613718411552
5,000,000+ : 6.825361010830325
500,000+ : 5.561823104693141
50,000+ : 4.7721119133574
5,000+ : 4.512635379061372
10+ : 3.5424187725631766
500+ : 3.2490974729241873
50,000,000+ : 2.3014440433213
100,000,000+ : 2.1322202166064983
50+ : 1.917870036101083
5+ : 0.78971119133574
1+ : 0.5076714801444043
500,000,000+ : 0.2707581227436823
1,000,000,000+ : 0.22563176895306858
0+ : 0.04512635379061372
0 : 0.01128158844765343


The install numbers aren't precise enough as most of the values are opened (100+, 5,000+, etc.). Since, we only want to get an idea of which app genres attract the most users, these categories satisfy our needs.

But, since we need these values as `floats`, the data needs to be cleaned.

In [25]:
android_categories = freq_table(android_free, 1)

avg_installs = {}
for category in android_categories:
    total_installs = 0
    n_apps_category = 0
    for app in android_free:
        if app[1] == category:
            installs = app[5]
            installs = installs.replace(',', '')
            installs = installs.replace('+', '')
            total_installs += float(installs)
            n_apps_category += 1
    avg_installs[category] = total_installs / n_apps_category
    
avg_installs = sorted(avg_installs.items(), key=lambda x: x[1], reverse=True)
for category, installs in avg_installs:
    print(category, ':', installs)

COMMUNICATION : 38456119.167247385
VIDEO_PLAYERS : 24727872.452830188
SOCIAL : 23253652.127118643
PHOTOGRAPHY : 17840110.40229885
PRODUCTIVITY : 16787331.344927534
GAME : 15588015.603248259
TRAVEL_AND_LOCAL : 13984077.710144928
ENTERTAINMENT : 11640705.88235294
TOOLS : 10801391.298666667
NEWS_AND_MAGAZINES : 9549178.467741935
BOOKS_AND_REFERENCE : 8767811.894736841
SHOPPING : 7036877.311557789
PERSONALIZATION : 5201482.6122448975
WEATHER : 5074486.197183099
HEALTH_AND_FITNESS : 4188821.9853479853
MAPS_AND_NAVIGATION : 4056941.7741935486
FAMILY : 3695641.8198090694
SPORTS : 3638640.1428571427
ART_AND_DESIGN : 1986335.0877192982
FOOD_AND_DRINK : 1924897.7363636363
EDUCATION : 1833495.145631068
BUSINESS : 1712290.1474201474
LIFESTYLE : 1437816.2687861272
FINANCE : 1387692.475609756
HOUSE_AND_HOME : 1331540.5616438356
DATING : 854028.8303030303
COMICS : 817657.2727272727
AUTO_AND_VEHICLES : 647317.8170731707
LIBRARIES_AND_DEMO : 638503.734939759
PARENTING : 542603.6206896552
BEAUTY : 51315

On average, communicatio apps have the most installs. Upon further investigation, this number is skewed due to apps such as WhatsApp, Facebook Messenger, Skype, Google Chrome, Gemail, and Hangouts that have over one billion installs as illustrated below.

In [26]:
for app in android_free:
    if app[1] == 'COMMUNICATION' and (app[5] == '1,000,000,000+' 
                                      or app[5] == '500,000,000+' 
                                      or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

WhatsApp Messenger : 1,000,000,000+
imo beta free calls and text : 100,000,000+
Android Messages : 100,000,000+
Google Duo - High Quality Video Calls : 500,000,000+
Messenger – Text and Video Chat for Free : 1,000,000,000+
imo free video calls and chat : 500,000,000+
Skype - free IM & video calls : 1,000,000,000+
Who : 100,000,000+
GO SMS Pro - Messenger, Free Themes, Emoji : 100,000,000+
LINE: Free Calls & Messages : 500,000,000+
Google Chrome: Fast & Secure : 1,000,000,000+
Firefox Browser fast & private : 100,000,000+
UC Browser - Fast Download Private & Secure : 500,000,000+
Gmail : 1,000,000,000+
Hangouts : 1,000,000,000+
Messenger Lite: Free Calls & Messages : 100,000,000+
Kik : 100,000,000+
KakaoTalk: Free Calls & Text : 100,000,000+
Opera Mini - fast web browser : 100,000,000+
Opera Browser: Fast and Secure : 100,000,000+
Telegram : 100,000,000+
Truecaller: Caller ID, SMS spam blocking & Dialer : 100,000,000+
UC Browser Mini -Tiny Fast Private & Secure : 100,000,000+
Viber Mess

The communication apps that have over 100 million installs will be removed to account for the skew.

In [27]:
under_100M = []

for app in android_free:
    installs = app[5]
    installs = installs.replace(',', '')
    installs = installs.replace('+', '')
    if app[1] == 'COMMUNICATION' and float(installs) < 100000000:
        under_100M.append(float(installs))

sum(under_100M) / len(under_100M)

3603485.3884615386

By removing the outliers, the average number of installs for communication apps decreases from 38,456,119 to 3,603,485.

There is a similar pattern in the video games category with strong players such as YouTube, Google Play Movies & TV, and MX Player, in the social category with strong players such as Facebook, Instagram, and Google+, in the photography category (Google Photos, etc.), and in the productivity category (Microsoft Word, Dropbox, Google Calendar, Evernote, etc.). Since these strong players are difficult to compete against, we will look at other categories.\

Again, the games category is popular, but seems to be saturated, so will look at other categories.

Next, the **books and reference** category looks popular with an average number of installs of 8,767,811. Lets explore this category in more depth.

In [28]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE':
        print(app[0], ':', app[5])

E-Book Read - Read Book for free : 50,000+
Download free book with green book : 100,000+
Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Free Panda Radio Music : 100,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
English Grammar Complete Handbook : 500,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
Google Play Books : 1,000,000,000+
AlReader -any text book reader : 5,000,000+
Offline English Dictionary : 100,000+
Offline: English to Tagalog Dictionary : 500,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
Recipes of Prophetic Medicine for free : 500,000+
ReadEra – free ebook reader : 1,000,000+
Anonymous caller detection : 10,000+
Ebook Reader : 5,000,000+
Litnet - E-books : 100,000+
Read books online : 5,000,000+
English to Urdu Dictionary : 500,000+
eBoox: book reader fb2 epub zip : 1,000,000+
English Persian Dictionary : 500,000+
Flybook : 500,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
E

Next, check for any apps that could skew our data set.

In [29]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000,000+'
                                           or app[5] == '500,000,000+'
                                           or app[5] == '100,000,000+'):
        print(app[0], ':', app[5])

Google Play Books : 1,000,000,000+
Bible : 100,000,000+
Amazon Kindle : 100,000,000+
Wattpad 📖 Free Books : 100,000,000+
Audiobooks from Audible : 100,000,000+


Looks like there are only a few very popular apps, so this market shows potential. Lets see what type of apps that show moderate popularity - between 1,000,000 and 100,000,000 downloads).

In [30]:
for app in android_free:
    if app[1] == 'BOOKS_AND_REFERENCE' and (app[5] == '1,000,000+'
                                           or app[5] == '5,000,000+'
                                           or app[5] == '10,000,000+'
                                           or app[5] == '50,000,000+'):
        print(app[0], ':', app[5])

Wikipedia : 10,000,000+
Cool Reader : 10,000,000+
Book store : 1,000,000+
FBReader: Favorite Book Reader : 10,000,000+
Free Books - Spirit Fanfiction and Stories : 1,000,000+
AlReader -any text book reader : 5,000,000+
FamilySearch Tree : 1,000,000+
Cloud of Books : 1,000,000+
ReadEra – free ebook reader : 1,000,000+
Ebook Reader : 5,000,000+
Read books online : 5,000,000+
eBoox: book reader fb2 epub zip : 1,000,000+
All Maths Formulas : 1,000,000+
Ancestry : 5,000,000+
HTC Help : 10,000,000+
Moon+ Reader : 10,000,000+
English-Myanmar Dictionary : 1,000,000+
Golden Dictionary (EN-AR) : 1,000,000+
All Language Translator Free : 1,000,000+
Aldiko Book Reader : 10,000,000+
Dictionary - WordWeb : 5,000,000+
50000 Free eBooks & Free AudioBooks : 5,000,000+
Al-Quran (Free) : 10,000,000+
Al Quran Indonesia : 10,000,000+
Al'Quran Bahasa Indonesia : 10,000,000+
Al Quran Al karim : 1,000,000+
Al Quran : EAlim - Translations & MP3 Offline : 5,000,000+
Koran Read &MP3 30 Juz Offline : 1,000,000+
H

The Books and Reference category seems to be dominated by software for processing and reading ebooks and various collections of libararies and dictionaries. Thus, its most likely not a good idea to build similar apps since there will be some significant competition.

There seems to be a few apps built around the book Quran, suggesting that building an app around a popular book can be profitable. Since the market is already full of libraries, it may be necessary to include additional features than just the book.

#### Apple Store - Avg number of user ratings per app genre

In [31]:
ios_genres = freq_table(ios_free, -5)

avg_ratings = {}
for genre in ios_genres:
    total_ratings = 0
    n_apps_genre = 0
    for app in ios_free:
        if app[-5] == genre:
            total_ratings += float(app[6])
            n_apps_genre += 1
    avg_ratings[genre] = total_ratings / n_apps_genre
    
avg_ratings = sorted(avg_ratings.items(), key=lambda x: x[1], reverse=True)
for genre, ratings in avg_ratings:
    print(genre, ':', ratings)

Navigation : 86090.33333333333
Reference : 74942.11111111111
Social Networking : 71548.34905660378
Music : 57326.530303030304
Weather : 52279.892857142855
Book : 39758.5
Food & Drink : 33333.92307692308
Finance : 31467.944444444445
Photo & Video : 28441.54375
Travel : 28243.8
Shopping : 26919.690476190477
Health & Fitness : 23298.015384615384
Sports : 23008.898550724636
Games : 22788.6696905016
News : 21248.023255813954
Productivity : 21028.410714285714
Utilities : 18684.456790123455
Lifestyle : 16485.764705882353
Entertainment : 14029.830708661417
Business : 7491.117647058823
Education : 7003.983050847458
Catalogs : 4004.0
Medical : 612.0


Navigation genre has the highest number of user reviews; however, this number is skewed due to Waze and Google Maps as illustrated below:

In [32]:
for app in ios_free:
    if app[-5] == 'Navigation':
        print(app[2], ':', app[6])

Waze - GPS Navigation, Maps & Real-time Traffic : 345046
Geocaching® : 12811
ImmobilienScout24: Real Estate Search in Germany : 187
Railway Route Search : 5
CoPilot GPS – Car Navigation & Offline Maps : 3582
Google Maps - Navigation & Transit : 154911


This same pattern is evident in the social netwokring and music genre, where they are dominated by Facebook, Pinterest, Skype, etc. and Pandora, Spotify, Shazam, etc., respectively.

Further the App Store is dominated by the 'fun' apps, indicating that a practical app might have more of a chance to stand out among the huge number of apps on the App Store.

Due to stiff competition, we will look at next popular genre, Reference.

In [33]:
for app in ios_free:
    if app[-5] == 'Reference':
        print(app[2], ':', app[6])

Bible : 985920
Dictionary.com Dictionary & Thesaurus : 200047
Dictionary.com Dictionary & Thesaurus for iPad : 54175
Muslim Pro: Ramadan 2017 Prayer Times, Azan, Quran : 18418
Merriam-Webster Dictionary : 16849
Google Translate : 26786
Night Sky : 12122
WWDC : 762
Jishokun-Japanese English Dictionary & Translator : 0
教えて!goo : 0
VPN Express : 14
New Furniture Mods - Pocket Wiki & Game Tools for Minecraft PC Edition : 17588
LUCKY BLOCK MOD ™ for Minecraft PC Edition - The Best Pocket Wiki & Mods Installer Tools : 4693
Guides for Pokémon GO - Pokemon GO News and Cheats : 826
Horror Maps for Minecraft PE - Download The Scariest Maps for Minecraft Pocket Edition (MCPE) Free : 718
City Maps for Minecraft PE - The Best Maps for Minecraft Pocket Edition (MCPE) : 8535
GUNS MODS for Minecraft PC Edition - Mods Tools : 1497
Real Bike Traffic Rider Virtual Reality Glasses : 8


Upon further speculation of the Reference genre, it's apparent that this genre is skewed by the Bible and Dictionary.com.

Regardless, this genres shows potential given that taking a popular book and turning into an app with additional features, such as embeding a dictionary within the app, could be promising.

Other genres that seem popular include weather, book, food and drink, or finance. The book genre seem to overlap a bit with the app idea we described above, but the other genres don't seem too interesting to us. For example, most people don't spend much time in weather apps, so the chances of making profit from in-app adds are low. Additionally, food and drink apps are from popular brands that provide delivery services, which is beyond the scope of this company. Lastly, finance apps requires domain knowledge, which is beyond the scope of this company as well.

## Conclusion


The goal of this project was to reommend an app profile that can be profitable in both the App Store and Google Play markets by analyzing data about their respective mobile apps.

It seems that taking a popular book (perhaps a more recent book) and turning it into an app could be profitable for both the Google Play and the App Store markets. Considering both markets are lready full of libraries, adding an additional feature will be critical to stand out. Examples of potential features include an audio version of the book, a forum for discussion or an embeded dictionary.